src/quoridor/game/game.go

```diff
@@ -10,11 +10,82 @@ import (
 
 // Game is the controller  
 type Game struct {
-	Id    string
+	ID    string `json:"id"`
 	Pawn  Pawn  `json:"pawn"`
+	Fences []Fence `json:"fences"`
 	Board *Board `json:"board"`
 }
 
+//AddFence add the fence on the board
+func (g Game) AddFence(fence Fence) error {
+	positionSquare := NewPositionSquare(fence.NWSquare)
+	if g.hasAlreadyAFenceAtTheSamePosition(fence.NWSquare) || g.hasNeighbourFence(fence.Horizontal, positionSquare) {
+		return errors.New("The fence overlaps another one")
+	}
+	fenceBlock := NewFenceBlock(fence.NWSquare)
+	g.addFenceWithEdges(fence, fenceBlock)
+	return nil
+}
+
+func (g Game) addFenceWithEdges(fence Fence, fenceBlock FenceBlock) error {
+	var firstEdge, secondEdge Edge 
+	if fence.Horizontal {
+		firstEdge = Edge{fenceBlock.NWSquare, fenceBlock.SWSquare} //westEdge
+		secondEdge = Edge{fenceBlock.NESquare, fenceBlock.SESquare} //eastEdge
+	} else {
+		firstEdge = Edge{fenceBlock.NWSquare, fenceBlock.NESquare} //northEdge
+		secondEdge = Edge{fenceBlock.SWSquare, fenceBlock.SESquare} //southEdge
+	}
+	err := g.addFenceIfCrossable(fence, firstEdge, secondEdge)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (g Game) addFenceIfCrossable(fence Fence, edge1 Edge, edge2 Edge) error {
+	if !g.isCrossableForAllPawns() {
+		return errors.New("No more access to goal line")
+	}
+	g.Fences = append(g.Fences, fence)
+	return nil
+}
+
+func (g Game) hasAlreadyAFenceAtTheSamePosition(p Position) bool {
+	for i := range g.Fences {
+		pos := g.Fences[i].NWSquare
+		if pos.Equals(p) {
+			return true
+		}
+	}
+	return false
+}
+
+func (g Game) hasNeighbourFence(isHorizontal bool, ps PositionSquare) bool {
+	if (isHorizontal) {
+		for i := range g.Fences {
+			fence := g.Fences[i]
+			pos := fence.NWSquare
+			if fence.Horizontal && (pos.Equals(ps.EastPosition) || pos.Equals(ps.WestPosition)) {
+				return true
+			}
+		}
+		return false
+	}
+	for i := range g.Fences {
+		fence := g.Fences[i]
+		pos := fence.NWSquare
+		if !fence.Horizontal && (pos.Equals(ps.NorthPosition) || pos.Equals(ps.SouthPosition)) {
+			return true
+		}
+	}
+	return false
+}
+
+func (g Game) isCrossableForAllPawns() bool {
+	return true
```

Comment:
utils?

Reply:
It was temporary, I thought

Reply:
Yes it is, just to add the logic, I know where to put my pathfinding